src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -37,8 +38,44 @@ private Currency getCurrency (String currencyString) {
         return Currency.valueOf(currencyString);
     }
 
-    private void processCommand () throws InvalidAddCommandException,
+    /**
+     * This method is used to validate the amount that is being passed in
+     * by the user and determines if it is a valid amount. If the amount
+     * provided is invalid, we will throw an error. Otherwise, we will
+     * return the amount to be processed.
+     *
+     * @param amount The amount that is being provided in string
+     * @return The big decimal amount that was being processed
+     * @throws NumberFormatException      If the string that is being processed
+     *                                    cannot be represented as a big decimal.
+     * @throws InvalidBigDecimalException If the amount provided is invalid.
+     *                                    This may be due to more than 2 dp or if
+     *                                    the value is more than 10_000_000.
+     */
+    private BigDecimal validateAndGetAmount (String amount) throws InvalidBigDecimalException, 
             InvalidAmountToAddException {
+        BigDecimal value = new BigDecimal(amount);
+
+        // Checks whether more than 2 dp is provided.
+        if (Math.max(0, value.stripTrailingZeros().scale()) > 2) {
+            throw new InvalidBigDecimalException("Please do not provide more than 2 decimal places");
+        }
+
+        // Checks whether the amount is more than 10_000_000. We do not compare directly as it will
+        // lead to an issue of overflow.
+        if (value.compareTo(new BigDecimal("10000000")) > 0) {
+            throw new InvalidBigDecimalException("Please do not provide a value of more than $10,000,000");
+        }
```

Comment:
For the convert function within Forex, I put a cap of 1bn because anything greater would cause implicit truncation. Should we try to standardize the cap we put on all commands or do it on a case-by-case basis?